stats/kafka/config_test.go

```diff
@@ -64,3 +69,55 @@ func TestConfigParseArg(t *testing.T) {
 	assert.Equal(t, null.StringFrom("influxdb"), c.Format)
 	assert.Equal(t, expInfluxConfig, c.InfluxDBConfig)
 }
+
+func TestConsolidatedConfig(t *testing.T) {
+	t.Parallel()
+	// TODO: add more cases
+	testCases := map[string]struct {
+		jsonRaw json.RawMessage
+		env     map[string]string
+		arg     string
+		config  Config
+		err     string
+	}{
+		"default": {
+			config: Config{
+				Format:         null.StringFrom("json"),
+				PushInterval:   types.NullDurationFrom(1 * time.Second),
+				InfluxDBConfig: influxdb.NewConfig(),
+			},
+		},
+		"bad influxdb concurrent writes": {
+			env: map[string]string{"K6_INFLUXDB_CONCURRENT_WRITES": "-2"},
+			config: Config{
+				Format:       null.StringFrom("json"),
+				PushInterval: types.NullDurationFrom(1 * time.Second),
+				InfluxDBConfig: influxdb.NewConfig().Apply(
+					influxdb.Config{
+						ConcurrentWrites: null.IntFrom(-2),
+					}),
+			},
+		},
+	}
+
+	for name, testCase := range testCases {
+		testCase := testCase
+		t.Run(name, func(t *testing.T) {
+			// hacks around env not actually being taken into account
+			os.Clearenv()
+			defer os.Clearenv()
+			for k, v := range testCase.env {
+				require.NoError(t, os.Setenv(k, v))
+			}
+
+			config, err := GetConsolidatedConfig(testCase.jsonRaw, testCase.env, testCase.arg)
+			if testCase.err != "" {
+				require.Error(t, err)
```

Comment:
There are no `err` cases to hit this, but OK :)

Reply:
I was really hoping that the "not positive concurrent writes" will break the config, but it breaks much later, and even with some refactoring I would've needed to also spin up kafka server to get the error ... which is *way* outside the scope of this PR or test